src/NHibernate/Engine/QueryParameters.cs

```diff
@@ -139,6 +157,11 @@ public bool HasRowSelection
 
 		public bool Callable { get; set; }
 
+		/// <summary>
+		/// Indicates if we can add loaded child collections to second level cache.
+		/// </summary>
+		public HashSet<string> UncacheableCollections { get; set; }
+
```

Comment:
Why can't we store this information right inside `Loader`? Looks like less changes to me (no need to modify `InitializeEntitiesAndCollections`)